client/src/OverviewComponents/ProductDetails.jsx

```diff
@@ -2,56 +2,65 @@ import React, { Component } from 'react';
 import { Button, Container, Row, Col } from 'react-bootstrap';
 import { Star } from 'react-bootstrap-icons';
 import Variants from './Variants.jsx';
+import Gallery from './Gallery.jsx';
 
 class ProductDetails extends React.Component {
   constructor(props) {
     super(props);
     this.state = { selectedStyle: '', selectedStyleImages: [] };
     this.onThumbnailClick = this.onThumbnailClick.bind(this);
+    this.updateStyleOptions = this.updateStyleOptions.bind(this);
+    this.selectedImages = [];
```

Comment:
line 13: i've never just created a variable in the constructor, has that been working well for you?